awslabs/amazon-timestream-tools
# Upgrade Flink Connector Dependency

Issue #, if available: N/A

N/A

Description of changes:

Upgrade flink dependency for Timestream Flink connector. Update documentation for running application with flink 1.18 on Managed Apache Flink. Revisions to `SinkInitContext.java` required for new API changes with the upgrade. The sample kinesis application jar now has all dependencies to execute without additional runtime configurations.

Validations:
- [x] - flink-connector-timestream compiles
  - Execute command: `mvn clean compile`
- [x] - flink-connector-timestream tests pass
  - Execute command: `mvn test`
- [x] - sample-kinesis-to-timestream-app compiles
  - Execute command: `mvn clean compile`
- [x] - sample-kinesis-to-timestream-app tests pass
  - Execute command: `mvn test`
- [x] - sample-kinesis-to-timestream-app  ran on Managed Apache Flink and flink 1.18 can ingest to Timestream
  - Complete the following:
    - `cd ../sample-kinesis-to-timestream-app`
    - `mvn clean compile && mvn package` 
    - `aws s3 cp target/sample-kinesis-to-timestream-app-0.1-SNAPSHOT.jar s3://YOUR_BUCKET_NAME/sample-kinesis-to-timestream-app-0.1-SNAPSHOT.jar`
    - Create application in Managed Apache Flink
    - Configure application as per README
    - Run application
    - Ingest data to kinesis stream using data generator
    - Validate data is ingested to Timestream without any application errors in Manage Apache Flink
- [x] - sample-kinesis-to-timestream-app ran locally can ingest to Timestream
  - Execute the following commands:
    - `mvn clean compile && mvn package`
    - `mvn install exec:java -Dexec.mainClass="com.amazonaws.samples.kinesis2timestream.StreamingJob" -Dexec.args="--InputStreamName TimestreamTestStream --Region us-east-1 --TimestreamDbName kdaflink --TimestreamTableName kinesisdata" -Dexec.classpathScope=test`
- [x] - sample-data-generator can ingest to kinesis stream
  - Execute command: `python3 kinesis_data_gen.py --stream TimestreamTestStream --region us-east-1`

## How to test

- Execute the following under `flink-connector-timestream`: `mvn clean compile && mvn package && mvn install`
- Update the `pom.xml` in `sample-kinesis-to-timestream-app` `flink-connector-timestream` dependency to the following:
```xml
        <dependency>
            <groupId>com.amazonaws.samples.connectors.timestream</groupId>
            <artifactId>flink-connector-timestream</artifactId>
            <version>0.5-SNAPSHOT</version>
        </dependency>
```
- Execute the following under `sample-kinesis-to-timestream-app` to build the jar: `mvn clean compile && mvn package`
- Execute the following under `sample`kinesis-to-timestream-app` to execute the jar file: `java -jar target/sample-kinesis-to-timestream-app-0.1-SNAPSHOT.jar --InputStreamName TimestreamTestStream --Region us-east-1 --TimestreamDbName kdaflink --TimestreamTableName kinesisdata`
- Execute the following under `sample-data-generator` to ingest data to the kinesis stream: `python3 kinesis_data_gen.py --stream TimestreamTestStream --region us-east-1`
- Validate data is being ingested to the `kinesisdata` table in the `kdaflink` database in region `us-east-1` 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
